### PR TITLE
Fix getting att pre state in sync

### LINKED ISSUE #6294: attestation fails to get pre state
# 🐞 Bug Report

### Description
Committee attestation's validation at sync attempts to get the prestate before verifying the attestation has prestate/should be in the pending queue

### Has this worked before in a previous version?
Yes

## 🔬 Minimal Reproduction
Run time with 1000 validators

## 🔥 Error

```
[2020-06-17 15:05:41] ERROR sync: Failed to retrieve pre state error=unknown state summary
could not get state summary
github.com/prysmaticlabs/prysm/beacon-chain/state/stategen.(*State).StateByRoot
	beacon-chain/state/stategen/getter.go:28
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).getAttPreState
	beacon-chain/blockchain/process_attestation_helpers.go:42
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).AttestationPreState
	beacon-chain/blockchain/receive_attestation.go:74
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).validateCommitteeIndexBeaconAttestation
	beacon-chain/sync/validate_beacon_attestation.go:78
github.com/prysmaticlabs/prysm/beacon-chain/sync.wrapAndReportValidation.func1
	beacon-chain/sync/subscriber.go:174
```

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**
> Bug fix

**What does this PR do? Why is it needed?**
The ordering for attestation's validation pipeline was wrong, the node should check pre state exists first before getting the pre state

**Which issues(s) does this PR fix?**

Fixes #6294 

**Other notes for review**
Tested run time

### CODE REVIEW

File: beacon-chain/sync/validate_beacon_attestation.go [hunk 1]
```diff
@@ -68,6 +68,17 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		return pubsub.ValidationIgnore
 	}
 
+	// Verify the block being voted and the processed state is in DB and. The block should have passed validation if it's in the DB.
```
Comment (shayzluf) [inline beacon-chain/sync/validate_beacon_attestation.go:71]: please change the validation order in the comment above as well

Comment (terencechain) [reply beacon-chain/sync/validate_beacon_attestation.go:71]: Huh? This comment is correct

Comment (terencechain) [reply beacon-chain/sync/validate_beacon_attestation.go:71]: Huh? This comment is correct

Comment (shayzluf) [reply beacon-chain/sync/validate_beacon_attestation.go:71]: this one
https://github.com/prysmaticlabs/prysm/blob/453434240b3b71602a277ab9145abe8d2f5a7c37/beacon-chain/sync/validate_beacon_attestation.go#L24

**Pull request merged**